### PR TITLE
80421 details in history in excel

### PR DESCRIPTION
**Pull request by janid1967**
Main task for PBI: Extend ExcelConverter to show preservations details in History for History records of type Requirement Preservation. Such History records have preservation details in an associated PreservationRecord

Found a column History.PreservationRecordId which is not used in DB. History records which is associated to a PreservationRecord, is connected via a Guid and not Id. Removed unused column. Also removed unused History constructor.

It became very inconvenient to test details in Excel converting in integrationtests. Added Unit tests for ExcelConverter, and moved all details testing on Excel creation from TagsControllerTests as unit tests

### CODE REVIEW

File: src/Equinor.Procosys.Preservation.Query/GetTagsQueries/GetTagsForExport/ExportHistoryDto.cs [hunk 1]
```diff
@@ -8,17 +8,24 @@ public class ExportHistoryDto
...
             DueInWeeks = dueInWeeks;
+            PreservationDetails = preservationDetails;
+            PreservationComment = preservationComment;
         }
 
         public int Id { get; }
         public string Description { get; }
         public DateTime CreatedAtUtc { get; }
         public int? DueInWeeks { get; }
+
```
Comment (bratli) [inline src/Equinor.Procosys.Preservation.Query/GetTagsQueries/GetTagsForExport/ExportHistoryDto.cs:27]: whitespace

File: src/Equinor.Procosys.Preservation.Query/GetTagsQueries/GetTagsForExport/ExportTagDto.cs [hunk 2]
```diff
@@ -56,6 +54,8 @@ public class ExportTagDto
             Status = status;
             Step = step;
             StorageArea = storageArea;
+
```
Comment (bratli) [inline src/Equinor.Procosys.Preservation.Query/GetTagsQueries/GetTagsForExport/ExportTagDto.cs:57]: whitespace

File: src/Equinor.Procosys.Preservation.Domain/AggregateModels/ProjectAggregate/PreservationRecord.cs [hunk 3]
```diff
@@ -27,6 +27,7 @@ public PreservationRecord(string plant, Person preservedBy, bool bulkPreserved)
             ObjectGuid = Guid.NewGuid();
```
Comment (bratli) [inline src/Equinor.Procosys.Preservation.Domain/AggregateModels/ProjectAggregate/PreservationRecord.cs:27]: unused CommentLengthMax in this file

**Review by bratli — APPROVED** [review]

File: src/Equinor.Procosys.Preservation.Query/GetTagsQueries/GetTagsForExport/ExportTagDto.cs (hunk 2 above)
Comment (janid1967) [reply src/Equinor.Procosys.Preservation.Query/GetTagsQueries/GetTagsForExport/ExportTagDto.cs:57]: wont fix

**Pull request merged**